http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/LifeCycleEventListener.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.transport.http.netty.contract;
+
+/**
+ * An event listener to listen to connector life cycle events.
+ */
+public interface LifeCycleEventListener {
```

Comment:
I am a bit concerned about the name here. Because this only has events related to port binding. But ideally, a connector life-cycle would have something like onCreate, onDestroy, onOpen, onClose. Therefore, I believe a better name would be something like PortBindingEeventListener. Later, in case we need other events we can change it something else.